### PR TITLE
SL: Rework Murmur Intro RP

### PR DESCRIPTION
**Pull request by miraco**
### Proof

- None

### Issues

- None

### How2Test

- None

### Todo / Checklist

- [X] None

### CODE REVIEW

**Comment by miraco** [comment]
Script should be ready for review now, everything else is handled via database.

File: src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp [hunk 1]
```diff
@@ -95,71 +121,45 @@ struct boss_murmurAI : public CombatAI
...
+            m_firstCast = true;
+        }
+        else
+        { 
+            // Kill a moving target
+            if (urand(0, 1))
             {
-                if (Creature* creature = m_creature->GetMap()->GetCreature(guid))
+                GuidVector m_WrathTargetGuid;
+                std::vector<Creature*> const* m_WrathTarget = m_creature->GetMap()->GetCreatures(MURMURS_WRATH_TARGETS_03);
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp:136]: local variable shouldnt be called m_ , thats for struct/class members

File: src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp [hunk 2]
```diff
@@ -95,71 +121,45 @@ struct boss_murmurAI : public CombatAI
...
+            DoCastSpellIfCan(nullptr, SPELL_SUPPRESSION_BLAST);
+            m_firstCast = true;
+        }
+        else
+        { 
+            // Kill a moving target
+            if (urand(0, 1))
             {
-                if (Creature* creature = m_creature->GetMap()->GetCreature(guid))
+                GuidVector m_WrathTargetGuid;
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp:135]: same here with the m_

File: src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp [hunk 3]
```diff
@@ -95,71 +121,45 @@ struct boss_murmurAI : public CombatAI
...
                 }
-            }
-            if (moversVector.size() > 0)
-            {
-                if (ObjectGuid& guid = moversVector[urand(0, moversVector.size() - 1)])
+
+                if (m_WrathTargetGuid.size() > 0)
                 {
-                    if (Creature* creature = m_creature->GetMap()->GetCreature(guid))
+                    if (ObjectGuid& guid = m_WrathTargetGuid[urand(0, m_WrathTargetGuid.size() - 1)])
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp:146]: Useless to have this in an if and also useless to use &. ObjectGuid is 64-bit and a ref is 64-bit. Would only make sense if you wanted to edit it after.

File: src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/shadow_labyrinth.cpp [hunk 4]
```diff
@@ -189,6 +189,57 @@ void instance_shadow_labyrinth::OnCreatureRespawn(Creature* creature)
...
+        case SL_BLACKHEAT_GROUP_05:
+        case SL_BLACKHEAT_GROUP_06:
+        case SL_BLACKHEAT_GROUP_07:
+        case SL_BLACKHEAT_GROUP_08:
+        case SL_BLACKHEAT_GROUP_09:
+        case SL_BLACKHEAT_GROUP_10:
+        case SL_BLACKHEAT_GROUP_11:
+        case SL_BLACKHEAT_GROUP_12:
+        {
+            switch (urand(0, 5))
```
Comment (miraco) [inline src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/shadow_labyrinth.cpp:214]: Chance is guessed, but after multiple tests on retail/cata i assume that every group in this room has a small chance to spawn assasins.
it was possible that killing 1 group spawned both assasins. 
There are always 
- 3 patrols (3 spawn groups)
- 2 group with 2 npc (left and right side under the tent)
- 2 static groups of 6 (1 left 1 right) 
- 2 groups with random spawns (middle left)
- 1 group with random spawns (middle right)
resulting in having 10 groups over all in this room.

File: src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp (hunk 3 above)
Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/outland/auchindoun/shadow_labyrinth/boss_murmur.cpp]: should be fixed

**Pull request merged**